### DISCUSSION TITLE
Cache management of micro-lc/orchestrator

### DISCUSSION THREAD

**Discussion by giovanni-motterle-mia-care** [mia-platform/community#556] [Q&A] [answered]
### What product are you having troubles with?

Microfrontend Composer

### What Console version are you using?

v13.1.1

### Description

Hi, I noted that when I'm navigating through Backoffice pages, orchestrator doesn't retrieve the page configuration again if since the last browser refresh I already visited that page.
In my use case configuration pages can change at runtime based on some "platform variables" and I was wondering if there was a way to disable this "internal" cache of orchestrator.

**Comment by epessina** [comment]
Hi @giovanni-motterle-mia-care , there is no internal cache in micro-lc or middleware. 

To avoid caching of configuration files you need to add the appropriate rules on your gateway to append the [Cache-Control](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control) header of the requests (specifically with value `no-cache` to disable cache entirely).

**Reply by giovanni-motterle-mia-care** [reply]
Hi, usually when no internal cache is present I can see an http call starting with the indication that the response is taken from the browser cache. Since in this case no new http calls are visualized I assumed the presence of an internal cache.
I will look into the cache control header you linked, thanks.

**Reply by giovanni-motterle-mia-care** [reply]
Hi, we configured the cache-control header but, as you can see in the video, microlc doesn't even try to fetch the configuration via http the second time i go to that page


https://github.com/user-attachments/assets/97b4166d-23d5-48a1-8fe6-d94e0d6fb698

**Reply by epessina** [reply]
Hi @giovanni-motterle-mia-care . Sorry, I didn't realize that was your problem. `micro-lc` (or, better yet, its internal orchestrator) does not re-fetch configurations of already visited pages. 

However, since it does make sense in some cases to have "fresh" configurations at each navigation, we can plan to support this feature

**Reply by giovanni-motterle-mia-care** [reply]
Hi, awesome, thanks.

**Reply by epessina** [reply, marked as answer]
Hi @giovanni-motterle-mia-care , we just released version `2.4.3` of micro-lc addressing the configuration re-fetching issue.

Starting from this version, when [configuring](https://micro-lc.io/docs/guides/applications/compose#usage) applications of type `compose` you can specify a new property `options` that may have a [property](https://micro-lc.io/docs/guides/applications/compose#fetchconfigonmount) `fetchConfigOnMount`. If this flag set to `true`, micro-lc will re-fetch the application configuration each time you visit it.

For example

```json
{
  "applications": {
    "refresh": {
       "integrationMode": "compose",
       "route": "./refresh",
       "config": "/micro-lc-configurations/refresh.config.json",
       "options": {
         "fetchConfigOnMount": true
       }
    }
  }
}
```

Please note that you cannot set this property via the application modal in the Configurator for now. You will need to manually set it in the advanced section.

Let me know if this implementation covers your use case

**Reply by giovanni-motterle-mia-care** [reply]
@epessina, awesome, thanks. I already configured it without issues.
